cuqi/experimental/mcmc/_cwmh.py

```diff
@@ -0,0 +1,221 @@
+        single independent sample from proposal distribution. Defaults to a
+        Gaussian proposal.  *Optional*.
+
+    scale : float or ndarray
+        Scale parameter used to define correlation between previous and proposed
+        sample in random-walk.  *Optional*. If float, the same scale is used for
+        all dimensions. If ndarray, a (possibly) different scale is used for
+        each dimension.
+
+    initial_point : ndarray
+        Initial parameters. *Optional*
+
+    callback : callable, *Optional*
+        If set this function will be called after every sample.
+        The signature of the callback function is
+        `callback(sample, sample_index)`, where `sample` is the current sample
+        and `sample_index` is the index of the sample.
+        An example is shown in demos/demo31_callback.py.
+
+    kwargs : dict
+        Additional keyword arguments to be passed to the base class 
+        :class:`ProposalBasedSamplerNew`.
+
+    Example
+    -------
+    .. code-block:: python
+        import numpy as np
+        import cuqi
+        # Parameters
+        dim = 5 # Dimension of distribution
+        mu = np.arange(dim) # Mean of Gaussian
+        std = 1 # standard deviation of Gaussian
+
+        # Logpdf function
+        logpdf_func = lambda x: -1/(std**2)*np.sum((x-mu)**2)
+
+        # Define distribution from logpdf as UserDefinedDistribution (sample
+        # and gradients also supported as inputs to UserDefinedDistribution)
+        target = cuqi.distribution.UserDefinedDistribution(
+            dim=dim, logpdf_func=logpdf_func)
+
+        # Set up sampler
+        sampler = cuqi.experimental.mcmc.CWMHNew(target, scale=1)
+
+        # Sample
+        samples = sampler.sample(2000).get_samples()
+
+    """
+    def __init__(self, target: cuqi.density.Density, proposal=None, scale=1,
+                 initial_point=None, **kwargs):
+        super().__init__(target, proposal=proposal, scale=scale,
+                         initial_point=initial_point, **kwargs)
+
+        # set initial scale
+        self.scale = scale
+
+        # set initial acceptance rate
+        self._acc = [np.ones((self.dim))]
+
+    @property
+    def scale(self):
+        """ Get the scale parameter. """
+        return self._scale
+
+    @scale.setter
+    def scale(self, value):
+        """ Set the scale parameter. """
+        if isinstance(value, Number):
+            self._scale = np.ones(self.dim)*value
+        elif isinstance(value, np.ndarray):
+            self._scale = value
+        self._scale_temp = self._scale.copy()
+
+    def validate_target(self):
+        if not isinstance(self.target, cuqi.density.Density):
+            raise ValueError(
+                "Target should be an instance of "+\
+                f"{cuqi.density.Density.__class__.__name__}")
+
+    @ProposalBasedSamplerNew.proposal.setter
+    # TODO. Check if we can refactor this.
```

Comment:
Do we have an issue to remember this? If not, add one?

Reply:
Thank you for pointing this out, I recorded it in #373